### PR TITLE
OpenCL: add rev support for (l)beta, cbrt, ceil, floor, round, trunc

### PR DESCRIPTION
**Pull request by rok-cesnovar**
## Summary

This PR adds OpenCL rev support for: 
- beta
- cbrt
- ceil
- floor
- lbeta
- round
- trunc

## Tests

All new functions are tested.

## Side Effects

/

## Release notes

Adds OpenCL rev support for beta, cbrt, ceil, floor, lbeta, round, trunc

## Checklist

- [x] Math issue #1854 

- [x] Copyright holder: Rok Češnovar

### CODE REVIEW

**Comment by rok-cesnovar** [comment]
Ready for review.

File: test/unit/math/opencl/rev/beta_test.cpp [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    = [](const auto& a, const auto& b) { return stan::math::beta(a, b); };
+
+TEST(OpenCL_beta, beta) {
+  Eigen::VectorXd in1(4);
+  in1 << 0.5, 3.4, 5.2, 7.5;
+  Eigen::VectorXd in2(4);
+  in2 << 3.3, 0.9, 6.7, 1.8;
+  stan::math::test::compare_cpu_opencl_prim_rev(beta_functor, in1, in2);
+}
+
```
Comment (t4c1) [inline test/unit/math/opencl/rev/beta_test.cpp:18]: Maybe also add a test on zero sized argument and a large one?

File: test/unit/math/opencl/rev/diag_post_multiply_test.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  return stan::math::diag_post_multiply(a, b);
+};
+
+TEST(OpenCL_diag_post_multiply, prim_rev_values_small) {
+  int N = 2;
+  int M = 3;
+
+  Eigen::MatrixXd a(N, M);
+  a << 1, 2, 3, 4, 5, 6;
+  Eigen::VectorXd b(M);
```
Comment (t4c1) [inline test/unit/math/opencl/rev/diag_post_multiply_test.cpp:17]: If you intend to support row vectors add tests for them.

File: test/unit/math/opencl/rev/lbeta_test.cpp [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    = [](const auto& a, const auto& b) { return stan::math::lbeta(a, b); };
+
+TEST(OpenCL_lbeta, value_small) {
+  Eigen::VectorXd in1(4);
+  in1 << 0.5, 3.4, 5.2, 7.5;
+  Eigen::VectorXd in2(4);
+  in2 << 3.3, 0.9, 6.7, 1.8;
+  stan::math::test::compare_cpu_opencl_prim_rev(lbeta_functor, in1, in2);
+}
+
```
Comment (t4c1) [inline test/unit/math/opencl/rev/lbeta_test.cpp:18]: Zero and large size tests?

**Review by t4c1 — CHANGES_REQUESTED** [review]

File: test/unit/math/opencl/rev/diag_post_multiply_test.cpp (hunk 2 above)
Comment (rok-cesnovar) [reply test/unit/math/opencl/rev/diag_post_multiply_test.cpp:17]: Not intended right now.

File: stan/math/opencl/prim/diag_post_multiply.hpp [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ * vector or row_vector.
+ */
+template <
+    typename T1, typename T2,
+    require_all_nonscalar_prim_or_rev_kernel_expression_t<T1, T2>* = nullptr>
+auto diag_post_multiply(const T1& m1, const T2& m2) {
+  check_vector("diag_post_multiply (OpenCL)", "m2", m2);
+  check_size_match("diag_post_multiply (OpenCL)", "m2.size()", m2.size(),
+                   "m1.cols()", m1.cols());
+  return m1 * diag_matrix(m2);
```
[the remark this answers is not in the document]
Comment (rok-cesnovar) [reply stan/math/opencl/prim/diag_post_multiply.hpp]: I agree that this isnt really efficient but this way it works with rev easily. rowwise_broadcast and elt_multiply do not have a /rev implementation so this needs a custom gradient now. Will check how easy/hard it is.

File: test/unit/math/opencl/rev/diag_post_multiply_test.cpp (hunk 2 above)
Comment (t4c1) [reply test/unit/math/opencl/rev/diag_post_multiply_test.cpp:17]: In that case fix the doxygen for diag_*_multiply, to not mention row vectors.

Comment (rok-cesnovar) [reply test/unit/math/opencl/rev/diag_post_multiply_test.cpp:17]: Nevermind, we need to support both, I misunderstood. CPU version supports both and so should this one. Or I will just scratch it for now.

**Comment by rok-cesnovar** [comment]
Decided to scratch `diag_{post, pre}_multiply` in this PR. Prim only support is not all that useful and I am just going to add both once I figure out the gradient.

**Review by t4c1 — APPROVED** [review]
